hmcts/pre-api
# S28 2911: DTO List Sort Orders

### Change description
- `BookingDTO` lists sorted
    - `participants`: sorted by participant first name
    - `captureSessions`: sorted by capture session id
    - `shares`: sorted by shared with user's first name
- `CaseDTO` lists sorted
    - `participants`: sorted by participant first name
- `CourtDTO` lists sorted
    - `regions`: sorted by region name
    - `rooms`: sorted by room name
- `RecordingDTO` lists sorted
    - `participants`: sorted by participant first name
- `UserDTO` list sorted
    - `appAccess`: sorted by last access (descending)
- `GET /bookings/{id}/share` returns sorted by `shared_with_user.first_name` (matching `BookingDTO.shares`)





> [!IMPORTANT]
> This PR requires manual testing by QA. Please notify the QA team @hmcts/pre-rec-evidence-qa.

**QA instructions**
- Ensure the sorting has actually been effective. 
- With GET /bookings call you can check the sorting of the BookingDTO responses are sorted
- With GET /cases call you can check the sorting of the CaseDTO response
- With GET /courts call you can check the sorting of the CourtDTO response
- With GET /recordings call you can check the sorting of the RecordingDTO response
- With GET /users call you can check the sorting of the UserDTO response
- With GET /bookings/{id}/share can check sorted by shared with user's first name
